### PR TITLE
FIX: Validate value of custom user fields (dropdowns and multiple selects)

### PR DESCRIPTION
**Pull request by jmperez127**
This PR is to address this issue from meta: https://meta.discourse.org/t/validation-of-dropdown-custom-user-fields/198132

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -147,7 +147,7 @@ def update
         field_id = f.id.to_s
         next unless params[:user_fields].has_key?(field_id)
 
-        val = params[:user_fields][field_id]
+        val = custom_field_value(f)
```
Comment (markvanlan) [inline app/controllers/users_controller.rb:150]: This is just my preference, and I know you didn't make it this way to start, but can you change `f` to `field` here? I really dislike abbreviations for short variables, they just make it so much harder for me to read!

**Review by markvanlan — APPROVED** [review]
I really like the specs here, nice work. Only leaving a small suggestion

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (jmperez127) [reply app/controllers/users_controller.rb:150]: I feel the same as you. Also `val` could be `value` ;)

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -1260,6 +1260,51 @@ def post_user(extra_params = {})
...
+
+          fab!(:user_field) do
+            Fabricate(:user_field, field_type: 'dropdown') do
+              user_field_options { [Fabricate(:user_field_option, value: "valid value")] }
+            end
+          end
+
+          it "shouldn't allow unregistered field values" do
+            expect do
+              put update_user_url, params: { user_fields: { field_id => "This option doesn't exists" } }
```
Comment (eviltrout) [inline spec/requests/users_controller_spec.rb:1277]:
```suggestion
              put update_user_url, params: { user_fields: { field_id => "This option doesn't exist" } }
```

**Review by eviltrout — APPROVED** [review]

**Review by eviltrout — APPROVED** [review]
Looks good!

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -143,16 +143,16 @@ def update
...
+      fields.each do |field|
+        field_id = field.id.to_s
         next unless params[:user_fields].has_key?(field_id)
 
-        val = params[:user_fields][field_id]
-        val = nil if val === "false"
-        val = val[0...UserField.max_length] if val
+        value = clean_custom_field_values(field)
+        value = nil if value === "false"
+        value = value[0...UserField.max_length] if value
```
Comment (markvanlan) [inline app/controllers/users_controller.rb:152]: I'm wondering if all this logic should be moved into `clean_custom_field_values`

Comment (jmperez127) [reply app/controllers/users_controller.rb:152]: I think we could, although the method would be slightly longer. I'm going to leave it here for simplicity, for now.

**Pull request merged**